robottelo/api/utils.py

```diff
@@ -940,42 +940,42 @@ def create_org_admin_user(orgs, locs):
     return user
 
 
-def skip_yum_update_during_provisioning(template=None, reverse=False):
+@contextmanager
+def skip_yum_update_during_provisioning(template=None):
     """Hides the yum update command with echo text
 
     :param str template: The template name where the yum update will be hidden
-    :param bool reverse: Reverses the echo text to yum update
-    :return: Boolean True on success else exception
     """
     old = 'yum -t -y update'
     new = 'echo "Yum update skipped for faster automation testing"'
-    if not reverse:
-        return update_provisioning_template(name=template, old=old, new=new)
-    else:
-        return update_provisioning_template(name=template, old=new, new=old)
+    update_provisioning_template(name=template, old=old, new=new)
+    yield
+    update_provisioning_template(name=template, old=new, new=old)
 
 
-def set_hammer_api_timeout(timeout=-1, reverse=False):
+@contextmanager
+def hammer_api_timeout(timeout=-1):
     """Set hammer API request timeout on Satellite
 
     :param int timeout: request timeout in seconds
-    :param bool reverse: Reverses the request timeout
-    :return: ssh.command
     """
-    default_timeout = f':request_timeout: {120}'
     new_timeout = f':request_timeout: {timeout}'
-    if not reverse:
-        return ssh.command(
-            "sed -ie 's/{}/{}/' ~/.hammer/cli.modules.d/foreman.yml".format(
-                default_timeout, new_timeout
-            )
-        )
+    cli_conf = "~/.hammer/cli.modules.d/foreman.yml"
+
+    if ssh.command(f"grep -i 'request_timeout' {cli_conf}").return_code != 0:
```

Comment:
will be good if we access using the satellite class object, for this PR I'm good.

Reply:
@shweta83 I would rather create a separate PR for ssh module, As I see this Pr intention was to quickly fix the test for CI failures. We are going to have have separate branch for SSH module as well.